src/tracker.js

```diff
@@ -1,147 +1,158 @@
+import { version } from "../package.json";
 
 export default class ShakaTracker extends nrvideo.VideoTracker {
-  setPlayer (player, tag) {
-    if (!tag && player.getMediaElement) tag = player.getMediaElement()
-    nrvideo.VideoTracker.prototype.setPlayer.call(this, player, tag)
+  setPlayer(player, tag) {
+    if (!tag && player.getMediaElement) tag = player.getMediaElement();
+    nrvideo.VideoTracker.prototype.setPlayer.call(this, player, tag);
   }
 
-  getTrackerName () {
-    return 'shaka'
+  getTrackerName() {
+    return "shaka";
   }
 
-  getTrackerVersion () {
-    return version
+  getTrackerVersion() {
+    return version;
   }
 
-  isLive () {
-    return this.player.isLive()
+  isLive() {
+    return this.player.isLive();
   }
 
-  getSrc () {
-    return this.player.getManifestUri()
+  getSrc() {
+    return this.player.getAssetUri();
   }
 
-  getPlayrate () {
-    return this.player.getPlaybackRate()
+  getPlayrate() {
+    return this.player.getPlaybackRate();
   }
 
-  getTrack () {
-    var tracks = this.player.getVariantTracks()
+  getTrack() {
+    var tracks = this.player.getVariantTracks();
     for (var i in tracks) {
-      var track = tracks[i]
-      if (track.active && (track.type === 'video' || track.type === 'variant')) {
-        return track
+      var track = tracks[i];
+      if (
+        track.active &&
+        (track.type === "video" || track.type === "variant")
+      ) {
+        return track;
       }
     }
-    return {}
+    return {};
   }
 
-  getLanguage () {
-    return this.getTrack().language
+  getLanguage() {
+    return this.getTrack().language;
   }
 
-  getRenditionName () {
-    return this.getTrack().label
+  getRenditionName() {
+    return this.getTrack().label;
   }
 
-  getRenditionBitrate () {
-    return this.getTrack().bandwidth
+  getRenditionBitrate() {
+    return this.getTrack().bandwidth;
   }
 
-  getRenditionWidth () {
-    return this.getTrack().width
+  getRenditionWidth() {
+    return this.getTrack().width;
   }
 
-  getRenditionHeight () {
-    return this.getTrack().height
+  getRenditionHeight() {
+    return this.getTrack().height;
   }
 
-  getPlayerVersion () {
-    return shaka.Player.version
+  getPlayerVersion() {
+    return shaka.Player.version;
   }
 
-  registerListeners () {
-    nrvideo.Log.debugCommonVideoEvents(this.tag)
+  registerListeners() {
+
+    nrvideo.Log.debugCommonVideoEvents(this.tag);
     nrvideo.Log.debugCommonVideoEvents(this.player, [
-      null, 'buffering', 'loading', 'adaptation', 'emsg', 'trackschanged', 'unloading',
-      'expirationupdated', 'largegap', 'texttrackvisibility', 'trackschanged'
-    ])
+      null,
+      "buffering",
+      "loading",
+      "adaptation",
+      "emsg",
+      "trackschanged",
+      "unloading",
+      "expirationupdated",
+      "largegap",
+      "texttrackvisibility",
+      "trackschanged",
+    ]);
 
-    this.tag.addEventListener('loadstart', this.onDownload.bind(this))
```

Comment:
Look for a better way to bind these events, we can prevent the repetition.

**Example**: we can move the event and function mapping to a separate (js/json) file and in JS file just run a loop to bind, unbind the events, that way maintain the mapping as well.

Reply:
I slightly agree, but still I think we can find a better way to keep the events lists separate somewhere. It will increase the readability of the code.